mountpoint-s3/src/data_cache/cache_directory.rs

```diff
@@ -30,65 +31,83 @@ pub enum ManagedCacheDirError {
 impl ManagedCacheDir {
     /// Create a new directory inside the provided parent path.
     /// If the directory already exists, it will be deleted before being recreated.
-    pub fn new_from_parent<P: AsRef<Path>>(parent_path: P) -> Result<Self, ManagedCacheDirError> {
+    pub fn new_from_parent_with_cache_key<P: AsRef<Path>, P2: AsRef<Path>>(
+        parent_path: P,
+        cache_key: P2,
+    ) -> Result<Self, ManagedCacheDirError> {
+        let wrapping_path = parent_path.as_ref().join("mountpoint-cache");
         let managed_cache_dir = Self {
-            managed_path: parent_path.as_ref().join("mountpoint-cache"),
+            path_with_cache_key: wrapping_path.join(cache_key),
```

Comment:
`join` can append nested directories or even replace with an absolute path. Should we check for it?

Reply:
Or is there a more specific method?

Reply:
Or rather use `Path` methods like `is_relative` and `components`